src/modules/commander/Commander.cpp

```diff
@@ -3784,51 +3786,71 @@ void Commander::estimator_check()
 		if (_estimator_selector_status_sub.copy(&estimator_selector_status)) {
 			if (estimator_selector_status.primary_instance != _estimator_status_sub.get_instance()) {
 				_estimator_status_sub.ChangeInstance(estimator_selector_status.primary_instance);
+				_estimator_status_flags_sub.ChangeInstance(estimator_selector_status.primary_instance);
 			}
 		}
 	}
 
-	if (_estimator_status_sub.update()) {
-		const estimator_status_s &estimator_status = _estimator_status_sub.get();
+	if (_estimator_status_flags_sub.update()) {
+		const estimator_status_flags_s &estimator_status_flags = _estimator_status_flags_sub.get();
 
-		// Check for a magnetomer fault and notify the user
-		const bool mag_fault = (estimator_status.control_mode_flags & (1 << estimator_status_s::CS_MAG_FAULT));
-		const bool gnss_heading_fault = (estimator_status.control_mode_flags & (1 << estimator_status_s::CS_GPS_YAW_FAULT));
+		if (estimator_status_flags.cs_inertial_dead_reckoning || estimator_status_flags.cs_wind_dead_reckoning) {
+			_status_flags.position_dead_reckoning = true;
+
+		} else {
+			_status_flags.position_dead_reckoning = false;
 
-		if (!mag_fault_prev && mag_fault) {
+			// position requirements (update if not dead reckoning)
+			bool gps             = estimator_status_flags.cs_gps;
+			bool optical_flow    = estimator_status_flags.cs_opt_flow;
+			bool vision_position = estimator_status_flags.cs_ev_pos;
+
+			_status_flags.position_reliant_on_gps             =  gps && !optical_flow && !vision_position;
```

Comment:
Is there a better way to latch onto which position sources we require?

Reply:
For example we only ever had GPS, if it disappears in flight in flight `estimator_status_flags.cs_gps` will no longer be true. How do we know `GPS` is required?

Reply:
The information regarding what's required is not stored anywhere AFAIK. EKF2 uses the data it gets if allowed by the aidmask param and that's all. I think your proposal here makes sense.